packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -479,86 +479,75 @@ class CupertinoDatePicker extends StatefulWidget {
-            final String dayOfMonth = localizations.datePickerDayOfMonth(longestDayOfMonth, wd);
-            if (longestText.length < dayOfMonth.length) {
-              longestText = dayOfMonth;
-            }
-          }
-        }
-      case _PickerColumnType.month:
-        for (int i = 1; i <= 12; i++) {
-          final String month = standaloneMonth
-              ? localizations.datePickerStandaloneMonth(i)
-              : localizations.datePickerMonth(i);
-          if (longestText.length < month.length) {
-            longestText = month;
-          }
+// Estimate the minimum width that each column needs to layout its content.
+static double _getColumnWidth(
+  _PickerColumnType columnType,
+  CupertinoLocalizations localizations,
+  BuildContext context,
+  bool showDayOfWeek, {
+  bool standaloneMonth = false,
+}) {
+  final List<String> longTexts = <String>[];
+
+  switch (columnType) {
+    case _PickerColumnType.date:
+      for (int i = 1; i <= 12; i++) {
+        final String date =
+            localizations.datePickerMediumDate(DateTime(2018, i, 25));
+        longTexts.add(date);
+      }
+    case _PickerColumnType.hour:
+      for (int i = 0; i < 24; i++) {
+        final String hour = localizations.datePickerHour(i);
+        longTexts.add(hour);
+      }
+    case _PickerColumnType.minute:
+      for (int i = 0; i < 60; i++) {
+        final String minute = localizations.datePickerMinute(i);
+        longTexts.add(minute);
+      }
+    case _PickerColumnType.dayPeriod:
+      longTexts.add(localizations.anteMeridiemAbbreviation);
+      longTexts.add(localizations.postMeridiemAbbreviation);
+    case _PickerColumnType.dayOfMonth:
+      int longestDayOfMonth = 1;
+      for (int i = 1; i <= 31; i++) {
+        final String dayOfMonth = localizations.datePickerDayOfMonth(i);
+        longTexts.add(dayOfMonth);
+        longestDayOfMonth = i;
+      }
+      if (showDayOfWeek) {
+        for (int wd = 1; wd < DateTime.daysPerWeek; wd++) {
+          final String dayOfMonth =
+              localizations.datePickerDayOfMonth(longestDayOfMonth, wd);
+          longTexts.add(dayOfMonth);
         }
-      case _PickerColumnType.year:
-        longestText = localizations.datePickerYear(2018);
-    }
+      }
+    case _PickerColumnType.month:
+      for (int i = 1; i <= 12; i++) {
+        final String month = standaloneMonth
+            ? localizations.datePickerStandaloneMonth(i)
+            : localizations.datePickerMonth(i);
+        longTexts.add(month);
+      }
+    case _PickerColumnType.year:
+      longTexts.add(localizations.datePickerYear(2018));
+  }
 
-    assert(longestText != '', 'column type is not appropriate');
+  assert(longTexts.isNotEmpty, 'column type is not appropriate');
 
-    return TextPainter.computeMaxIntrinsicWidth(
+  final List<String> texts = (List<String>.from(longTexts)
+      ..sort((String a, String b) => b.length.compareTo(a.length)))
+      .take(4)
```

Comment:
Why 4 specifically?

Reply:
I considered that, on average, alphabetic and syllabic languages ​​tend to have 2-3 month names that are relatively long, while logographic and ideographic languages ​​also tend to have about 2 month names that are longer. Since abbreviations generally do not vary much in character count, I decided to select only 4 words for optimization.

I can't say for sure, but I think it's plausible.

Reply:
> tend to have about 2 month names that are longer

You can use a font with wide glyphs for certain code points. As a pathological example, I can make one character 5 times as wide as other characters, or make the advance of a character sequence much smaller with kerning or making it a ligature.

Reply:
Okay, I'll try to do that.